dla_cnn/desi/preprocess.py

```diff
@@ -219,3 +219,41 @@ def estimate_s2n(sightline):
     s2n = sightline.flux/sightline.error
     #return s/n
     return np.median(s2n[test])
+
+def generate_summary_table(sightlines, output_dir, mode = "w"):
+    """
+    Generate a csv file to store some necessary information of the given sightlines. The necessary information means the id, z_qso,
+    s/n of thelymann forest part(avoid dlas+- 3000km/s), the wavelength range and corresponding pixel number of each channel. And the csv file's format is like:
+    
+    id(int)， z_qso(float), s2n(float), wavelength_start_b(float), wavelength_end_b(float), pixel_start_b(int), pixel_end_b(int), wavelength_start_r(float), wavelength_end_r(float), pixel_start_r(int), pixel_end_r(int), wavelength_start_z(float), wavelength_end_z(float), pixel_start_z(int), pixel_end_z(int)
+
+    "wavelength_start_b" means the start wavelength value of b channel, "wavelength_end_b" means the end wavelength value of b channel, "pixel_start_b" means the start pixel number of b channel, "pixel_end_b" means the end pixel number of b channel
+    so do the other two channels.
+    -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+    parameters:
+    sightlines: list of `dla_cnn.data_model.Sightline.Sightline` object, the sightline contained should contain the all data of b,r,z channel, and shouldn't be rebinned,
+    output_dir: str, where the output csv file is stored, its format should be "xxxx.csv",
+    mode: str, possible values "w", "a", "w" means writing to the csv file directly(overwrite the previous content), "a" means adding more data to the csv file(remaining the previous content)
+    --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+    return:
+    None
+    
+    """
+    import csv
```

Comment:
put this (and all imports) at the top of the module

Reply:
OK , I have updated that. Thanks!